src/qtism/data/storage/xml/marshalling/AssessmentResultMarshaller.php

```diff
@@ -0,0 +1,124 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ * Copyright (c) 2018 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ * @author Moyon Camille, <camille@taotesting.com>
+ * @license GPLv2
+ */
+
+namespace qtism\data\storage\xml\marshalling;
+
+use DOMElement;
+use qtism\data\QtiComponent;
+use qtism\data\results\AssessmentResult;
+use qtism\data\results\ItemResultCollection;
+
+/**
+ * Class AssessmentResultMarshaller
+ *
+ * The marshaller to manage serialization between QTI component and DOM Element
+ * 
+ * @package qtism\data\storage\xml\marshalling
+ */
+class AssessmentResultMarshaller extends Marshaller
+{
+    /**
+     * Marshall a QtiComponent object into its QTI-XML equivalent.
+     *
+     * @param QtiComponent|AssessmentResult $component A QtiComponent object to marshall.
+     * @return DOMElement A DOMElement object.
+     * @throws MarshallingException
+     */
+    protected function marshall(QtiComponent $component)
+    {
+        $element = self::getDOMCradle()->createElement($this->getExpectedQtiClassName());
+
+        $context = $component->getContext();
+        $element->appendChild($this->getMarshallerFactory()->createMarshaller($context)->marshall($context));
+
+        if ($component->hasTestResult()) {
+            $testResult = $component->getTestResult();
+            $element->appendChild($this->getMarshallerFactory()->createMarshaller($testResult)->marshall($testResult));
+        }
+
+        if ($component->hasItemResults()) {
+            foreach ($component->getItemResults() as $itemResult) {
+                $element->appendChild($this->getMarshallerFactory()->createMarshaller($itemResult)->marshall($itemResult));
+            }
+        }
+
+        return $element;
+    }
+
+    /**
+     * Unmarshall a DOMElement object corresponding to a QTI sessionIdentifier element.
+     *
+     * @param DOMElement $element A DOMElement object.
+     * @return QtiComponent A QtiComponent object.
+     * @throws UnmarshallingException
+     */
+    protected function unmarshall(DOMElement $element)
+    {
+        try {
+            $contextElements = self::getChildElementsByTagName($element, 'context');
```

Comment:
I think this should be `$this->getChildElementsByTagName` and not `self::getChildElementsByTagName` because the signature of the getChildElementsByTagName is not static